pkg/lib/config/secret.go

```diff
@@ -86,26 +86,33 @@ func (c *SecretConfig) Overlay(layers ...*SecretConfig) *SecretConfig {
 	return merged
 }
 
-func (c *SecretConfig) Lookup(key SecretKey) (*SecretItem, bool) {
-	for _, item := range c.Secrets {
+func (c *SecretConfig) Lookup(key SecretKey) (*SecretItem, bool, int) {
+	for index, item := range c.Secrets {
 		if item.Key == key {
-			return &item, true
+			return &item, true, index
 		}
 	}
-	return nil, false
+	return nil, false, -1
 }
 
 func (c *SecretConfig) LookupData(key SecretKey) SecretItemData {
-	if item, ok := c.Lookup(key); ok {
+	if item, ok, _ := c.Lookup(key); ok {
 		return item.Data
 	}
 	return nil
 }
 
+func (c *SecretConfig) LookupDataWithIndex(key SecretKey) (SecretItemData, int) {
```

Comment:
It is better to add a last bool in the return value.

Reply:
Fixed